MyOnboarding/test/MyOnboardingApp.Services.Tests/Services/UpdateItemServiceTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using MyOnboardingApp.Contracts.Errors;
+using MyOnboardingApp.Contracts.Generators;
+using MyOnboardingApp.Contracts.Models;
+using MyOnboardingApp.Contracts.Repository;
+using MyOnboardingApp.Contracts.Validation;
+using MyOnboardingApp.Services.Services;
+using MyOnboardingApp.TestUtils.Extensions;
+using MyOnboardingApp.TestUtils.Factories;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace MyOnboardingApp.Services.Tests.Services
+{
+    [TestFixture]
+    [SuppressMessage("ReSharper", "IdentifierTypo")]
+    public class UpdateItemServiceTests
+    {
+        [Test]
+        public async Task EditItemAsync_ExistingItemSpecified_ReturnsEditedItemWithNoErrors()
+        {
+            var repository = Substitute.For<ITodoListRepository>();
+            var dateTimeGenerator = Substitute.For<IDateTimeGenerator>();
+            var validator = Substitute.For<IInvariantValidator<TodoListItem>>();
+            var service = new UpdateItemService(repository, dateTimeGenerator, validator);
+
+            var updatedTime = new DateTime(2018, 10, 10);
+            dateTimeGenerator
+                .GetCurrentDateTime()
+                .Returns(updatedTime);
+            var itemId = new Guid("00000000-0000-0000-0000-000000000001");
+            var (_, existingResolvedItem, _) = ItemVariantsFactory.CreateItemVariants(
+                itemId,
+                "old text",
+                new DateTime(2015, 01, 01),
+                new DateTime(2015, 02, 02)
+            );
+
+            var (newItem, _, _) = ItemVariantsFactory.CreateItemVariants(
+                itemId,
+                "new text");
+
+            var (_, _, expectedResult) = ItemVariantsFactory.CreateItemVariants(
+                itemId,
+                newItem.Text,
+                existingResolvedItem.Item.CreationTime,
+                updatedTime);
+
+           
+            repository
+                .ReplaceItemAsync(Arg.Any<TodoListItem>())
+                .Returns(callInfo => callInfo.Arg<TodoListItem>());
+            validator
+                .Validate(Arg.Any<TodoListItem>())
+                .Returns(callInfo => ItemWithErrors.Create(callInfo.Arg<TodoListItem>(), new List<Error>().AsReadOnly()));
```

Comment:
```suggestion
                .Returns(callInfo => ItemWithErrors.Create(callInfo.Arg<TodoListItem>(), Enumerable.Empty<Error>()));
```
(based on https://github.com/petrareb/kentico-onboarding-cs/pull/4#discussion_r263424985)